### ISSUE TITLE
Add pre-commit hook configuration for other projects to use

### ISSUE THREAD

**Issue by JustinKuli** [IBM/tekton-lint#62]
I recently started using [pre-commit](https://pre-commit.com/#intro) to manage the installation and running of linters in one of my projects. Basically (read their docs for more info) developers on projects can save references to linters/checkers/etc in a `.pre-commit-config.yaml` file in their repository, and new developers can get them installed more quickly and consistently - plus they can be configured to run on every git commit.

Some of the linters "supported" by pre-commit are in mirrors of the main projects (for example: [prettier](https://github.com/pre-commit/mirrors-prettier)), but other projects have added the required hook definitions to their repositories directly (for example: [go-jsonnet](https://github.com/google/go-jsonnet/blob/master/.pre-commit-hooks.yaml)). I think it would be great if we supported using tekton-lint through pre-commit.

To "support" using pre-commit with this project, we would just need to add a `.pre-commit-hooks.yaml` file, with [some specific information](https://pre-commit.com/#new-hooks). I've made a simple fork to test this out, and it looks good - although more tests might be needed: https://github.com/JustinKuli/tekton-lint/blob/main/.pre-commit-hooks.yaml. I'd be happy to open a PR based on that if there is interest.

To be clear, I'm not proposing that this project should start using/enforcing use of pre-commit in its own development workflow. This issue is focused on making tekton-lint easier for other projects to use via pre-commit.

**Comment by JustinKuli** [comment]
Also, I should note, it is already possible to "use" pre-commit to run tekton-lint in a project through a "local" hook. For example:
```yaml
# .pre-commit-config.yaml
repos:
  - repo: local
    hooks:
      - id: tekton-lint
        name: tekton-lint
        entry: tekton-lint
        language: system
        files: pipeline.yaml
```

However, this requires the developer to have installed pre-commit and tekton-lint separately. If this project had a `.pre-commit-hooks.yaml` file, then pre-commit could handle the installation of tekton-lint for the developer, along with any other linters/etc they were using through pre-commit.
